metals/src/main/scala/scala/meta/internal/metals/Tables.scala

```diff
@@ -35,20 +38,41 @@ final class Tables(
   val fingerprints =
     new Fingerprints(() => connection)
 
-  def connect(): Unit = {
-    this._connection =
-      if (clientConfig.initialConfig.isAutoServer) tryAutoServer()
-      else tryAutoServer()
+  private val ref: AtomicReference[ConnectionState] =
+    new AtomicReference(ConnectionState.Empty)
+
+  def connect(): Connection = {
+    ref.get() match {
+      case empty @ ConnectionState.Empty =>
+        if (ref.compareAndSet(empty, ConnectionState.InProgress)) {
+          val conn = tryAutoServer()
+          ref.set(ConnectionState.Connected(conn))
+          conn
+        } else
+          connect()
```

Comment:
Isn't it the same case as `InProgress` and we should wait a bit too?

Reply:
We will end up at Tables.ConnectionState.InProgress  right away, so it should be ok